terminus/tests/road_node_test.py

```diff
@@ -0,0 +1,15 @@
+import unittest
+
+from models.road_node import RoadNode
+
+from geometry.bounding_box import BoundingBox
+
+from geometry.point import Point
+
+
+class RoadNodeTest(unittest.TestCase):
+
+    def test_road_node_bounding_box(self):
```

Comment:
> Why would that happen?
That is a fair question and in the most common scenario because of a bug in the code calling it. However the aim of my question is more to think about all that comes together with defining a method, what to test and what not, etc. Couple of things to consider:

- Going back to "Why would that happen?" I may also say, "because it can happen". So the fact that it doesn't make sense, doesn't mean it can't happen. That is an important thing to always keep in mind.
- "I suppose it would return..." that is what tests are for, so that you (and later the user of your class) doesn't have to suppose, but get a well defined behavior. And it is important to mention that sometimes there is not a single answer to this; you could throw an error, take the absolute value of the width, decide to do a `max(0, width)` inside your implementation or let the bounding box be created with those values (which IIRC, due to the normalization, it will flip the origin and corner). The important thing is that you get a well defined behavior and your tests cover all the cases, even the "wrong" ones.